commit 3a913fa535c185d017625b352a32d1643d1f489e
Author: Anshuman Gupta <anshuman.gupta@intel.com>
Date:   Fri Mar 19 15:32:08 2021 +0530

    drm/i915/hdcp: return correct error code
    
    hdcp2_enable_stream_encryption shouldn't get called in case
    of any port authentication or encryption error, though
    hdcp2_enable_stream_encryption checks for link encryption
    before enabling stream encryption and returns error but
    this return error code won't be correct in case of any error
    due to port authentication and encryption.
    
    Cc: Ramalingam C <ramalingam.c@intel.com>
    Reviewed-by: Ankit Nautiyal <ankit.k.nautiyal@intel.com>
    Signed-off-by: Anshuman Gupta <anshuman.gupta@intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20210319100208.5886-4-anshuman.gupta@intel.com

diff --git a/drivers/gpu/drm/i915/display/intel_hdcp.c b/drivers/gpu/drm/i915/display/intel_hdcp.c
index 77650677ba6f..9248536e4906 100644
--- a/drivers/gpu/drm/i915/display/intel_hdcp.c
+++ b/drivers/gpu/drm/i915/display/intel_hdcp.c
@@ -1911,7 +1911,8 @@ static int hdcp2_authenticate_and_encrypt(struct intel_connector *connector)
 		}
 	}
 
-	ret = hdcp2_enable_stream_encryption(connector);
+	if (!ret)
+		ret = hdcp2_enable_stream_encryption(connector);
 
 	return ret;
 }